### ISSUE TITLE
[BUG] conditionalRowStyles is overwritten by striped rows

### ISSUE THREAD

**Issue by wrbutros** [jbetancur/react-data-table-component#651]
I'm not sure if this is actually a bug, but here we go:
I have a table with 'striped' rows enabled, but additional to that I would like to customize the row style depending on some conditions, for example is a given row is or not enabled.

By using conditionalRowStyle, only the rows which are not 'stripped' apply my `color` customized style.
The 'stripped' rows still apply other style attributes, such as `hover`.
For example, this conditional should apply the  `color: 'rgba(0,0,0,.54)'` to every row.

```
conditionalRowStyles: [{
      when: () => true,
      style: {
        color: 'rgba(0,0,0,.54)',
        '&:hover': {
          cursor: 'not-allowed',
        },
      },
    }]
```
But only the not stripped row get that color applied.
By other way, both rows change the cursor as 'not-allowed' when it is `hover`.

![image](https://user-images.githubusercontent.com/3587379/90065469-54071700-dcba-11ea-9fbc-f3e769e09e20.png)
![image](https://user-images.githubusercontent.com/3587379/90065498-5ec1ac00-dcba-11ea-8857-3eaa0ba6dac8.png)

**Comment by jbetancur** [comment]
Yep this is totally a bug. I'll see about getting a fix in soon.

**Comment by jbetancur** [comment]
The issue is that the striped applies a very specific rule nth-of-type(odd)  so when you set the style it gets overridden by the most specific style. A workaround for you in the meantime is to also override this in your conditional styles

For example
```js
  {
    when: row => row.calories < 300,
    style: {
      backgroundColor: 'rgba(63, 195, 128, 0.9)',
      color: 'white',
      '&:hover': {
        cursor: 'pointer',
      },
      '&:nth-of-type(odd)': {
        color: 'white,
        backgroundColor: rgba(63, 195, 128, 0.9)',
      },
    },
  },

```

This fix for this unfortunately requires a bit of a rethink/refactor on how row striping is done. Which is really more of a convenience thing

**Comment by jbetancur** [comment]
Fix was easier than I though. I removed the CSS nth-of-type(odd) and replaced this with a modulo function to determine if the row is odd or not. The conditional styles can now properly override the CSS selectors. 6.11.1 has the fix.
